tests/components/speedtestdotnet/test_sensor.py

```diff
@@ -0,0 +1,41 @@
+"""Tests for SpeedTest sensors."""
+from homeassistant.components import speedtestdotnet
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.components.speedtestdotnet.const import DEFAULT_NAME, SENSOR_TYPES
+
+from . import MOCK_RESULTS, MOCK_SERVERS
+
+from tests.async_mock import patch
+from tests.common import MockConfigEntry
+
+
+async def test_speedtestdotnet_sensors(hass):
+    """Test sensors created for speedtestdotnet integration."""
+    entry = MockConfigEntry(domain=speedtestdotnet.DOMAIN, data={})
+    entry.add_to_hass(hass)
+
+    with patch("speedtest.Speedtest") as mock_api:
+        mock_api.return_value.get_best_server.return_value = MOCK_SERVERS["Server1"][0]
+        mock_api.return_value.results.dict.return_value = MOCK_RESULTS
+
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+        assert len(hass.states.async_entity_ids(SENSOR_DOMAIN)) == 3
+
+        for sensor_type in SENSOR_TYPES:
+            if sensor_type == "ping":
```

Comment:
Set the correct result in a dictionary so we don't need this check, but can just use the `sensor_type` to look up the expected result for assertion.